internal/osbuildexecutor/runner-impl-aws-ec2.go

```diff
@@ -0,0 +1,97 @@
+package osbuildexecutor
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"os/exec"
+
+	"github.com/osbuild/images/pkg/osbuild"
+	"github.com/sirupsen/logrus"
+
+	"github.com/osbuild/osbuild-composer/internal/cloud/awscloud"
+)
+
+type awsEC2Executor struct {
+	iamProfile string
+}
+
+func (ec2e *awsEC2Executor) RunOSBuild(manifest []byte, store, outputDirectory string, exports, exportPaths, checkpoints,
+	extraEnv []string, result bool, errorWriter io.Writer) (*osbuild.Result, error) {
+	region, err := awscloud.RegionFromInstanceMetadata()
+	if err != nil {
+		return nil, err
+	}
+
+	aws, err := awscloud.NewDefault(region)
+	if err != nil {
+		return nil, err
+	}
+
+	si, err := aws.RunSecureInstance(ec2e.iamProfile)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := aws.TerminateSecureInstance(si)
+		if err != nil {
+			logrus.Errorf("Error terminating secure instance: %v", err)
+		}
+	}()
+
+	logrus.Info("Spinning up jobsite manager")
+	args := []string{
+		"--builder-host",
+		*si.Instance.PrivateIpAddress,
+	}
+
+	for _, exp := range exports {
+		args = append(args, "--export", exp)
+	}
+	for _, exp := range exportPaths {
+		args = append(args, "--export-file", exp)
+	}
+	for _, env := range extraEnv {
+		args = append(args, "--environment", env)
+	}
+	args = append(args, "--output", outputDirectory)
+
+	cmd := exec.Command(
```

Comment:
I think this should have a timeout context, just in case something gets stuck. I'm not sure how long it should be though, so I'd err on the side of longer to make sure it doesn't accidentally interrupt a running build.

Reply:
So the `/usr/libexec/osbuild-composer/osbuild-jobsite-manager` already had a bunch of timeouts included inside of it. That waits 60 minutes.

Reply:
ok, as long as you think those are reliable then this should be ok.

Reply:
I'll keep an eye on it, if they turn out not to be we can of course add it.